crates/astria-sequencer/src/bridge/bridge_lock_action.rs

```diff
@@ -226,18 +241,100 @@ mod tests {
+            ..default_deposit()
+        });
+        assert_correct_base_deposit_fee(&Deposit {
+            asset: "test_asset".parse().unwrap(),
+            ..default_deposit()
+        });
+        assert_correct_base_deposit_fee(&Deposit {
+            destination_chain_address: "someaddresslonger".to_string(),
+            ..default_deposit()
+        });
+
+        // Ensure calculated length is as expected with absurd string
+        // lengths (have tested up to 99999999, but this makes testing very slow)
+        let absurd_string: String = ['a'; u16::MAX as usize].iter().collect();
+        assert_correct_base_deposit_fee(&Deposit {
+            asset: absurd_string.parse().unwrap(),
+            ..default_deposit()
+        });
+        assert_correct_base_deposit_fee(&Deposit {
+            destination_chain_address: absurd_string,
+            ..default_deposit()
+        });
+    }
+
+    #[track_caller]
+    #[allow(clippy::arithmetic_side_effects)] // allow: test will never overflow u128
+    fn assert_correct_base_deposit_fee(deposit: &Deposit) {
+        let calculated_len = calculate_base_deposit_fee(deposit).unwrap();
+        let expected_len = DEPOSIT_BASE_FEE
+            + deposit.asset.to_string().len() as u128
+            + deposit.destination_chain_address.len() as u128;
+        assert_eq!(calculated_len, expected_len);
+    }
+
+    /// Used to determine the base deposit byte length for `get_deposit_byte_len()`. This is based
+    /// on "reasonable" values for all fields except `asset` and `destination_chain_address`. These
+    /// are empty strings, whose length will be added to the base cost at the time of
+    /// calculation.
+    ///
+    /// This test determines 165 bytes for an average deposit with empty `asset` and
+    /// `destination_chain_address`, which is divided by 10 to get our base byte length of 16. This
+    /// is to allow for more flexibility in overall fees (we have more flexibility multiplying by a
+    /// lower number, and if we want fees to be higher we can just raise the multiplier).
+    #[test]
+    fn get_base_deposit_fee() {
+        use prost::Message as _;
+        let bridge_address = Address::builder()
+            .prefix("astria-bridge")
+            .slice(&[0u8; ADDRESS_LEN][..])
+            .try_build()
+            .unwrap();
+        let raw_deposit = astria_core::generated::sequencerblock::v1alpha1::Deposit {
+            bridge_address: Some(bridge_address.to_raw()),
+            rollup_id: Some(RollupId::from_unhashed_bytes([0; ROLLUP_ID_LEN]).to_raw()),
+            amount: Some(1000.into()),
+            asset: String::new(),
+            destination_chain_address: String::new(),
+            source_transaction_id: Some(TransactionId::new([0; TRANSACTION_ID_LEN]).to_raw()),
+            source_action_index: 0,
+        };
+        assert_eq!(DEPOSIT_BASE_FEE, raw_deposit.encoded_len() as u128 / 10);
+    }
+
+    fn default_deposit() -> Deposit {
```

Comment:
It's not really default, maybe:
```suggestion
    fn reference_deposit() -> Deposit {
```